test cases/common/234 generator target/gen1/meson.build

```diff
@@ -0,0 +1,12 @@
+g1 = generator(find_program('comp2xml.py'),
+  output: '@BASENAME@.xc',
+  arguments: ['@INPUT@', '@OUTPUT@'],
+)
+
+xml1 = generator_target('gen1',
+  input: ['first.component',
+          'second.component',
+          'third.component',
+         ],
+  generator: g1
+)
```

Comment:
Trying to understand the goal here, how does this make things better than using a `custom_target()` with multiple outputs? Is the point that the tool (`comp2xml.py`) does not need to handle multiple inputs in one invocation?

Reply:
> how does this make things better than using a custom_target() with multiple outputs

A custom target consists of only one external process call. Most generator programs can not handle multiple inputs and outputs and must instead be called once per input file. This is that, basically "like a generator, but output everything to the current build dir".